client/rest/src/main/java/org/opensearch/client/RestClient.java

```diff
@@ -954,6 +954,34 @@ public InputStream getContent() throws IOException {
             }
             return out.asInput();
         }
+
+        /**
+         * A gzip compressing entity doesn't work with chunked encoding with sigv4
+         *
+         * @return false
+         */
+        @Override
+        public boolean isChunked() {
+            return false;
+        }
+
+        /**
+         * A gzip entity requires content length in http headers
+         * as it doesn't work with chunked encoding for sigv4
+         *
+         * @return content length of gzip entity
+         */
+        @Override
+        public long getContentLength() {
+            long size;
+            try (InputStream is = getContent()) {
+                size = is.readAllBytes().length;
+            } catch (IOException ex) {
+                size = -1L;
```

Comment:
Is there other places where we return `-1` when we cannot figure out the content length. I would expect an exception, not a negative number causing all kinds of wonderful problems of doing `x.size() + content.getContentLength()`.

Reply:
I see https://github.com/apache/httpcomponents-core/blob/master/httpcore5/src/main/java/org/apache/hc/core5/http/protocol/RequestContent.java#L115 so looks like it's a thing.

Reply:
yes that why I think @reta said to return -1.
I don't see any other place where `we're doing x.size + content.getContentLength();`
or returning -1

Reply:
It's fine. The interface would have been "unsigned long" if it was meant to be otherwise.